relay.go

```diff
@@ -394,3 +420,26 @@ func frameTypeFor(f *Frame) frameType {
 func errUnknownGroup(group string) error {
 	return NewSystemError(ErrCodeDeclined, "no peers for %q", group)
 }
+
+func determinesCallSuccess(f *Frame) (succeeded bool, failed bool, failMsg string) {
```

Comment:
optional: what do you think about using `len(failMsg) > 0` as `failed`? explicit is fine too, just means we have to be careful about making sure that when we receive a `failMsg`, we're also returning `failed = true`